mpqp/tools/circuit.py

```diff
@@ -0,0 +1,89 @@
+import random
+from typing import Optional
+
+import numpy as np
+
+from mpqp.core.circuit import QCircuit
+from mpqp.core.instruction.gates.gate import SingleQubitGate
+from mpqp.core.instruction.gates.native_gates import NATIVE_GATES, TOF, Rk, U
+from mpqp.core.instruction.gates.parametrized_gate import ParametrizedGate
+
+
+def random_circuit(
+    gate_classes: Optional[list[type]] = None,
+    nb_qubits: int = 5,
+    nb_gates: int = np.random.randint(5, 10),
+):
+    """This function creates a QCircuit with a specified number of qubits and gates.
+    The gates are chosen randomly from the provided list of native gate classes.
+
+    args:
+        nb_qubits : Number of qubits in the circuit.
+        gate_classes : List of native gate classes to use in the circuit.
+        nb_gates : Number of gates to add to the circuit. Defaults to a random
+         integer between 5 and 10.
+
+    Returns:
+        A quantum circuit with the specified number of qubits and randomly chosen gates.
+
+    Raises:
+        ValueError: If the number of qubits is too low for the specified gates.
+
+    Examples:
+        >>> random_circuit([U, TOF], 3) # doctest: +SKIP
+        Generates a random quantum circuit with 3 qubits using U and TOF gates.
```

Comment:
Why do we output a description of what it generates ? Can't we print the circuit, and still skip the doctest? (I understand it is hard to doctest random circuit generation)

Reply:
done

Reply:
we could also set the random seed for this test (or even for all tests) so that the result become stable

Reply:
Should we create a task on trello ?

Reply:
done